src/java.desktop/unix/classes/sun/font/XRGlyphCacheEntry.java

```diff
@@ -69,28 +69,25 @@ public int getGlyphSet() {
     }
 
     public static int getGlyphID(long glyphInfoPtr) {
-        // We need to access the GlyphID with Unsafe.getAddress() because the
+        // We need to access the GlyphID as a long because the
         // corresponding field in the underlying C data-structure is of type
         // 'void*' (see field 'cellInfo' of struct 'GlyphInfo'
         // in src/share/native/sun/font/fontscalerdefs.h).
         // On 64-bit Big-endian architectures it would be wrong to access this
-        // field with Unsafe.getInt().
-        return (int) StrikeCache.unsafe.getAddress(glyphInfoPtr +
-                                                   StrikeCache.cacheCellOffset);
+        // field as an int.
+        return (int)StrikeCache.getGlyphCellInfo(glyphInfoPtr);
     }
 
     public static void setGlyphID(long glyphInfoPtr, int id) {
-        // We need to access the GlyphID with Unsafe.putAddress() because the
+        // We need to access the GlyphID as a long because the
         // corresponding field in the underlying C data-structure is of type
-        // 'void*' (see field 'cellInfo' of struct 'GlyphInfo' in
-        // src/share/native/sun/font/fontscalerdefs.h).
-        // On 64-bit Big-endian architectures it would be wrong to write this
-        // field with Unsafe.putInt() because it is also accessed from native
-        // code as a 'long'.
+        // 'void*' (see field 'cellInfo' of struct 'GlyphInfo'
+        // in src/share/native/sun/font/fontscalerdefs.h).
+        // On 64-bit Big-endian architectures it would be wrong to access this
```

Comment:
"wrong to write this field as an int"?

Reply:
"access" because it is true for read and write.